lightning/src/ln/channelmanager.rs

```diff
@@ -3797,6 +3803,15 @@ impl<'a, ChanSigner: ChannelKeys + Readable, M: Deref, T: Deref, K: Deref, F: De
 			per_peer_state.insert(peer_pubkey, Mutex::new(peer_state));
 		}
 
+        let event_count: u64 = Readable::read(reader)?;
+        let mut pending_events_read: Vec<events::Event> = Vec::with_capacity(cmp::min(event_count as usize, mem::size_of::<events::Event>()));
```

Comment:
Heh, we can have more than one :p. Usually we define `const MAX_ALLOC_SIZE: u64 = 64*1024;` and then Probably instead of mem::size_of use MAX_ALLOC_SIZE / mem::size_of...

Reply:
I did notice for the other fields, the cmp::min is either the count variable or a hard coded 128 value. Should I create that variable let `MAX_ALLOC_SIZE: u64 = 54*1024;` and then use that for the other values being read in as well? 
We would end up with something like this:
`let mut pending_events_read: Vec<events::Event> = Vec::with_capacity(cmp::min(event count as usize, MAX_ALLOC_SIZE / mem::size_of::<events::Event>())))`